### PR TITLE
Test improvements: covered some possible execution path not yet tested

### PR DESCRIPTION
**Pull request by DonCallisto**
This is a draft PR.
Sneaking around, trying to improve test coverage.
I just stop for tests that's not unit tests, but other work can be done in order to cover more and more cases.

I left here for review. If good, as coverage is now more than 90% (that's a very good measure), we can directly merge it and open singular issues that can be picked by others.

WDYT?

### CODE REVIEW

File: tests/Badge/Model/UseCase/CreateErroreBadgeTest.php [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    public function testCreateAGenericErrorBadge(): void
+    {
+        $error = new \Exception('msg');
+        $format = 'svg';
+
+        $createErrorBadge = new CreateErrorBadge();
+
+        $badge = $createErrorBadge->createErrorBadge($error, $format);
+
+        $this->assertEquals(new Badge((string) $error, 'generic', 'e05d44', $format), $badge);
```
Comment (JellyBellyDev) [inline tests/Badge/Model/UseCase/CreateErroreBadgeTest.php:21]: `self::assertEquals()`?

File: tests/Badge/Model/UseCase/CreateErroreBadgeTest.php [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        $badResponseEx = $this->getMockBuilder(BadResponseException::class)
+            ->disableOriginalConstructor()
+            ->getMock();
+        $format = 'svg';
+
+        $createErrorBadge = new CreateErrorBadge();
+
+        $badge = $createErrorBadge->createErrorBadge($badResponseEx, $format);
+
+        $this->assertEquals(new Badge((string) $badResponseEx, 'not found?', 'e05d44', $format), $badge);
```
Comment (JellyBellyDev) [inline tests/Badge/Model/UseCase/CreateErroreBadgeTest.php:35]: `self::assertEquals()`?

File: tests/Badge/Model/UseCase/CreateErroreBadgeTest.php [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+    public function testCreateAClientExceptionErrorBadge(): void
+    {
+        $unexpectedValueEx = new \UnexpectedValueException('msg');
+        $format = 'svg';
+
+        $createErrorBadge = new CreateErrorBadge();
+
+        $badge = $createErrorBadge->createErrorBadge($unexpectedValueEx, $format);
+
+        $this->assertEquals(new Badge((string) $unexpectedValueEx, 'connection', 'e05d44', $format), $badge);
```
Comment (JellyBellyDev) [inline tests/Badge/Model/UseCase/CreateErroreBadgeTest.php:47]: `self::assertEquals()`?

File: tests/Badge/Model/UseCase/CreateVersionBadgeTest.php [hunk 4]
```diff
@@ -39,33 +40,92 @@ public function testShouldCreateLicenseBadge(): void
...
+            ->method('getLatestStableVersion')
+            ->willReturn('v2.0');
+
+        $this->repository->expects($this->any())
+            ->method('fetchByRepository')
+            ->willReturn($package);
+
+        $badge = $this->useCase->createStableBadge('PUGX/badge-poser');
+
+        $this->assertEquals(new Badge('stable', 'v2.0', '28a3df'), $badge);
```
Comment (JellyBellyDev) [inline tests/Badge/Model/UseCase/CreateVersionBadgeTest.php:52]: do not saw it anymore! :)

**Review by JellyBellyDev — CHANGES_REQUESTED** [review]
You are awesome!!!  
In my opinion we can merge this! The coverage with 90% it's very good! 
Later if anyone wants to add more coverage they are free to do so!

File: tests/Badge/Model/UseCase/CreateErroreBadgeTest.php (hunk 1 above)
Comment (DonCallisto) [reply tests/Badge/Model/UseCase/CreateErroreBadgeTest.php:21]: Why? Isn't this irrelevant?

File: tests/Badge/Model/UseCase/CreateErroreBadgeTest.php (hunk 2 above)
Comment (DonCallisto) [reply tests/Badge/Model/UseCase/CreateErroreBadgeTest.php:35]: Same

File: tests/Badge/Model/UseCase/CreateErroreBadgeTest.php (hunk 3 above)
Comment (DonCallisto) [reply tests/Badge/Model/UseCase/CreateErroreBadgeTest.php:47]: Same

File: tests/Badge/Model/UseCase/CreateVersionBadgeTest.php (hunk 4 above)
Comment (DonCallisto) [reply tests/Badge/Model/UseCase/CreateVersionBadgeTest.php:52]: I don't get it :sweat_smile:

File: tests/Badge/Model/UseCase/CreateErroreBadgeTest.php (hunk 1 above)
Comment (JellyBellyDev) [reply tests/Badge/Model/UseCase/CreateErroreBadgeTest.php:21]: Yes! But it's a new best practice! No? :)

Comment (DonCallisto) [reply tests/Badge/Model/UseCase/CreateErroreBadgeTest.php:21]: > A common question, especially from developers new to PHPUnit, is whether using $this->assertTrue() or self::assertTrue(), for instance, is "the right way" to invoke an assertion. The short answer is: there is no right way. And there is no wrong way, either. It is a matter of personal preference.

Source: https://phpunit.de/manual/6.5/en/appendixes.assertions.html (Appendix A. Assertions)

Comment (JellyBellyDev) [reply tests/Badge/Model/UseCase/CreateErroreBadgeTest.php:21]: Yep! I know! But for me there is a micro-optimization to use a method static! 
#my2cent

I leave the choice to you! ;)

**Review by JellyBellyDev — APPROVED** [review]
LGTM!

**Review by AlessandroMinoccheri — APPROVED** [review]
Amazing! LGTM 👍

**Pull request merged**